core/App/screens/ProofRequest.tsx

```diff
@@ -194,11 +196,21 @@ const ProofRequest: React.FC<ProofRequestProps> = ({ navigation, route }) => {
       ...retrievedCredentials?.requestedPredicates,
     }
 
-    // TODO:(jl) Need to test with partial match? Maybe `.some` would work?
-    return (
-      typeof retrievedCredentials !== 'undefined' &&
-      (credName ? fields[credName]?.length > 0 : Object.values(fields).every((c) => c.length > 0))
-    )
+    if (credName) {
+      let credFound = false
+      Object.keys(fields).forEach((proofKey) => {
+        const credNamesInAttrs = fields[proofKey].map((attr) =>
+          parseCredDefFromId(attr.credentialInfo?.credentialDefinitionId, attr.credentialInfo?.schemaId)
+        )
+        if (credNamesInAttrs.includes(startCase(credName))) {
+          credFound = true
+          return
+        }
+      })
+      return credFound
+    }
+
+    return !typeof retrievedCredentials && Object.values(fields).every((c) => c.length > 0)
```

Comment:
should this just be `return retrievedCredentials && Object.values(fields).every((c) => c.length > 0)`?

the `typeof` will give a non-empty string in almost every case, no?

Reply:
yes, my bad

Reply:
okay fixed it